prog/dftb+/lib_solvation/cm5.F90

```diff
@@ -0,0 +1,508 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2020  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+
+!> Implementation of the charge model 5 (CM5)
+module dftbp_cm5
+  use dftbp_assert
+  use dftbp_accuracy, only : dp
+  use dftbp_atomicrad, only : getAtomicRad
+  use dftbp_blasroutines, only : gemv
+  use dftbp_constants, only : AA__Bohr, symbolToNumber
+  use dftbp_periodic, only : TNeighbourList, getNrOfNeighboursForAll
+  use dftbp_simplealgebra, only : determinant33
+  implicit none
+  private
+
+  public :: TChargeModel5, TCM5Input, init
```

Comment:
Other modules are starting to use a naming convention of chargeModel5_int to give more meaningful compiler messages when there are mistakes in calling arguments (instead of the 'no matching interface' if the routine choice is defined by signature).

Reply:
How about `initChargeModel5` it nicely fits with the usual naming conventions for procedures.

Reply:
Personally, I'd vote for `TChargeModel5_init()`. I think, it makes it more readable, We already use underscore in names for indicating name spaces (e.g. in module names). IMHO, the straightforward convention would be `TypeName_typeRelatedRoutine`, as already used for several cases (see for example `initprogram.F90`).